build-tools/assign_grant.sh

```diff
@@ -1,15 +1,22 @@
 #!/bin/bash
 
+set -e
+
+if [ -f build-tools/grant_id.txt ]; then
+    >&2 echo "Grant already assigned! (Did you mean to 'revoke-grant' instead?)"
+    exit 1
+fi
+
 echo "********************"
 echo "* Assigning Grants *"
 echo "********************"
 
 account=$(aws sts get-caller-identity --query 'Account' --output text)
-account_arn=$(aws iam list-roles | grep "Lambda" | grep "arn" | awk '{print $2}' | tr -d \")
+account_arn=$(aws iam list-roles | grep "busy-engineers-ee-iam-LambdaRole-" | grep "arn" | awk '{print $2}' | tr -d \" | tr -d ",")
```

Comment:
Are we sure that the role is named with busy-engineers-ee-iam-LambdaRole? 
This might mess us up if it is not named that way for sure!

Reply:
Yes, I've tested it against a couple of setups of the workshop. https://github.com/aws-samples/busy-engineers-encryption-sdk/pull/158 this morning set up the default name for this prefix to make sure it matches this assumption elsewhere.

My concern is that "Lambda" is generic, and that this could mistakenly grant to some principal we don't expect, at which point I think it'll be harder to debug as well. If it bombs out because it can't find it, it'll fail at assignment time rather than runtime, and hopefully it'll be easier to figure out if the attendee changed the template name or something and go from there.

Reply:
Cool got it